omod/src/main/java/org/openmrs/module/reportingrest/web/controller/ReportingRestController.java

```diff
@@ -28,11 +57,139 @@ public class ReportingRestController extends MainResourceController {
 
     public static final String REPORTING_REST_NAMESPACE = "/reportingrest";
 
+    private static final Log LOGGER = LogFactory.getLog(ReportingRestController.class);
+
     /**
      * @see org.openmrs.module.webservices.rest.web.v1_0.controller.BaseRestController#getNamespace()
      */
     @Override
     public String getNamespace() {
         return RestConstants.VERSION_1 + REPORTING_REST_NAMESPACE;
     }
+
+    @RequestMapping(value = "/runReport", method = RequestMethod.POST)
+    @ResponseStatus(HttpStatus.OK)
+    public void runReport(@RequestBody RunReportRequest runReportRequest) {
```

Comment:
Current `ReportRequestResource` just saves a `ReportRequest` in DB, as-is.
The `runReport` creates `ReportRequest` and adds it to execution queue - maybe better name would be `queueReport` as from `ReportService.queueReport`.

Generally, this is migration of `org.openmrs.module.reportingui.page.controller.RunReportPageController`.

Would it make sense for `ReportRequestResource` to queue every new `ReportRequest`? 
We were hesitant to modify anything already in use.

Reply:
@pwargulak - It's been a long time since I've looked at this, but my first reaction is that it definitely would make the most sense for the `ReportRequestResource` to queue every new `ReportRequest`.  So basically:
* If the report request does not already exist with the given uuid, call `reportService.queueReport(ReportRequest)`, otherwise calls `reportService.saveReportRequest(ReportRequest);`